lib/rules/no-sparse-arrays.js

```diff
@@ -36,11 +38,25 @@ module.exports = {
         return {
 
             ArrayExpression(node) {
+                const { sourceCode } = context;
+                let commaToken;
+
+
+                for (const element of node.elements) {
+                    commaToken = sourceCode.getTokenAfter(
+                        element ?? commaToken ?? sourceCode.getFirstToken(node),
+                        astUtils.isCommaToken
+                    );
 
-                const emptySpot = node.elements.includes(null);
```

Comment:
Perhaps we could keep this quick check for an early exit (`if (!node.elements.includes(null)) { return; }`) since the rule now takes considerably more time due to searching for every comma?

This is TIMING for this repo before & after the change:

```
before:

no-sparse-arrays                                        |     5.912 |     0.0%

after:

no-sparse-arrays                                        |    49.170 |     0.2%

```